10up/classifai
# fix/759: Add background processing for Embeddings Classification

### Description of the Change


This PR implements Action Scheduler to perform Embeddings Classification with 20 terms per job. (The count is opinionated, open to change it).


Closes #759 


### How to test the Change


1. Visit `/wp-admin/tools.php?page=classifai&tab=language_processing`
2. Click save.
3. Observe the notice "**OpenAI Embeddings**: Classification is in progress in the background."
4. Wait until message updates to "**OpenAI Embeddings**: Classification completed."
5. Verify the embeddings generated in the `wp_termmeta` table.

### Changelog Entry

> Added - `classifai_openai_embeddings_max_terms` filter hook to change the number of terms to process during Embeddings classification per job.
> Changed - Embeddings classification now uses background processing to improve performance.


### Credits

Props @dkotter @Sidsector9 


### Checklist:
